components/Chat/Chat.vue

```diff
@@ -751,10 +751,15 @@
         voice.rate   = this.getTtsRate / 10.0;
         voice.volume = this.getTtsVolume / 10.0;
         voice.pitch  = pitch;
-        // Completely sanitized messages (i.e. emotes only) don't get read
-        // TODO: implement empty message filter
         voice.text   = ( this.getTtsReadUsername ? `${username} says: ` : '' ) + message;
 
+        // Completely sanitized messages (i.e. emotes only) don't get read
+        const pattern = new RegExp( '\S', 'gi' );
+        if( !message.match( pattern ) ) {
```

Comment:
you've used a capital S which is match *not* whitespace, when I think you want to use `\s+` to match at least 1 whitespace character and I think you need to ensure there is nothing else but whitespace by matching from the start `^` to the end `$` so the regex should probably be
`/^\s+$/` to match a string that has at least 1 whitespace character between the start and end, and no characters are valid except whitespace in the string

Reply:
Also to save an extra regex call during non-tts usage, you could add `this.getTtsReadUsername` to th eif statement because there's no need for this check unless we are reading the username. And same for the string construction, we can save the string concatenation on line 754 by prepending the username to voice.text = username + voice.text i think in the if statement as well, and setting voice.text to just message initially. or only in an if...else... whatever is the easiest to read. Definitely skip an extra regex if possible though.

Reply:
I had the same thoughts on minimising regex use, so the `\S` was intentional. It will match non-whitespaces, meaning there's some (presumably) readable text in the message. However, since I'm looking for empty messages (i.e. without readable text), I logically negated that match. Should be O(n), if match is implemented to stop at first match. 

I'll add the extra check to skip the regex, and think of a way to lighten it up more.